solidity/contracts/VerifierModule.sol

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.19;
+
+import {IStorageMirror} from 'interfaces/IStorageMirror.sol';
+import {MerklePatriciaProofVerifier} from 'libraries/MerklePatriciaProofVerifier.sol';
+import {StateVerifier} from 'libraries/StateVerifier.sol';
+import {RLPReader} from 'solidity-rlp/contracts/RLPReader.sol';
+import {IStorageMirrorRootRegistry} from 'interfaces/IStorageMirrorRootRegistry.sol';
+import {IBlockHeaderOracle} from 'interfaces/IBlockHeaderOracle.sol';
+import {IVerifierModule} from 'interfaces/IVerifierModule.sol';
+import {ISafe} from 'interfaces/ISafe.sol';
+import {Enum} from 'safe-contracts/common/Enum.sol';
+
+/**
+ * @title VerifierModule
+ * @notice This contract is the verifier module that verifies the settings of a safe against the StorageMirror on the home chain
+ */
+contract VerifierModule is IVerifierModule {
+  using RLPReader for RLPReader.RLPItem;
+  using RLPReader for bytes;
+
+  /**
+   * @notice The start of the linked list for the owners of a safe
+   * @dev Used for updating the owners of a safe
+   */
+
+  address internal constant _SENTINEL_OWNERS = address(0x1);
+
+  /**
+   * @notice The slot of the mapping of the safe to the keccak256 hash of the latest verified settings in the StorageMirror
+   */
+
+  uint256 internal constant _LATEST_VERIFIED_SETTINGS_SLOT = 0;
+
+  /**
+   * @notice The interface of the StorageMirrorRootRegistry contract
+   */
+
+  IStorageMirrorRootRegistry public immutable STORAGE_MIRROR_ROOT_REGISTRY;
+
+  /**
+   * @notice The interface of the block header oracle contract
+   */
+
+  IBlockHeaderOracle public immutable BLOCK_HEADER_ORACLE;
+
+  /**
+   * @notice The address of the StorageMirror contract on the home chain
+   */
+
+  address public immutable STORAGE_MIRROR;
+
+  /**
+   * @notice The mapping of the safe to the keccak256 hash of the latest verified settings
+   */
+
+  mapping(address => bytes32) public latestVerifiedSettings;
+
+  /**
+   * @notice The mapping of the safe to the timestamp of when the settings where verified
+   */
+
+  mapping(address => uint256) public latestVerifiedSettingsTimestamp;
+
+  constructor(address _storageMirrorRootRegistry, address _storageMirror, address _blockHeaderOracle) payable {
```

Comment:
Let's add them as interfaces directly not `address` except the storage mirror i guess which we only need the address